CMakeLists.txt

```diff
@@ -151,14 +152,28 @@ if(FUSILLI_BUILD_TESTS OR FUSILLI_BUILD_BENCHMARKS)
   # Add library for tests/utils.h
   add_library(libutils INTERFACE)
   target_include_directories(libutils INTERFACE ${CMAKE_CURRENT_SOURCE_DIR}/tests)
+  target_link_libraries(libutils INTERFACE
+    libfusilli
+    Catch2::Catch2WithMain
+  )
```

Comment:
I would have assumed that using `Catch2::Catch2WithMain` before `find_package(Catch2 3)` (or equivalent) would cause an error. But I guess not 🤷‍♂️ .

Reply:
Yeah it appears CMake doesn't immediately validate that link targets exist during the `target_link_libraries` call, probably defers target resolution until later. But this is fragile, let me refactor this a bit.

Reply:
Done.